File: composeApp/src/commonMain/kotlin/com/example/coveragex/presentation/screens/userlist/UserListState.kt
package com.example.coveragex.presentation.screens.userlist

import com.example.coveragex.domain.model.User

/**
 * UI state for the user list screen.
 */
sealed class UserListState {
    /**
     * Initial state before any data is loaded
     */
    data object Initial : UserListState()

    /**
     * Loading state while fetching data
     */
    data object Loading : UserListState()

    /**
     * Success state with user data
     * @param users All users from API
     * @param filteredUsers Users after applying search filter
     * @param isRefreshing Whether pull-to-refresh is active
     */
    data class Success(
        val users: List<User>,
        val filteredUsers: List<User>,
        val isRefreshing: Boolean = false
    ) : UserListState()

    /**
     * Error state when data fetch fails
     * @param message Error message to display
     */
    data class Error(val message: String) : UserListState()
}

File: iosApp/iosApp/Presentation/Screens/UserList/UserListState.swift
import Foundation
import Shared

/// UI state for the user list screen
/// Mirrors the Kotlin UserListState from shared module
enum UserListState: Equatable {
    case initial
    case loading
    case success(users: [User], filteredUsers: [User], isRefreshing: Bool)
    case error(message: String)
    
    static func == (lhs: UserListState, rhs: UserListState) -> Bool {
        switch (lhs, rhs) {
        case (.initial, .initial):
            return true
        case (.loading, .loading):
            return true
        case let (.success(users1, filtered1, refresh1), .success(users2, filtered2, refresh2)):
            return users1.map { $0.id } == users2.map { $0.id } &&
                   filtered1.map { $0.id } == filtered2.map { $0.id } &&
                   refresh1 == refresh2
        case let (.error(msg1), .error(msg2)):
            return msg1 == msg2
        default:
            return false
        }
    }
}
